src/components/Pressable/Pressable.tsx

```diff
@@ -0,0 +1,269 @@
+      props.onPress?.(adaptTouchEvent(event));
+    }
+
+    isPressedDown.current = false;
+    setPressedState(false);
+  }, []);
+
+  // fix for: touch out is called before touch in due to async .measure() in onTouchesDown()
+  const handlingOnTouchesDown = useRef<boolean>(false);
+  const onEndHandlingTouchesDown = useRef<() => void>(() => null);
+
+  const touchGesture = useMemo(
+    () =>
+      Gesture.Manual()
+        .onTouchesDown((event) => {
+          handlingOnTouchesDown.current = true;
+          pressableRef.current?.measure((_x, _y, width, height) => {
+            // pressableRef.current?._nativeTag yields targetId on android & ios
+            if (
+              !isTouchWithinInset(
+                {
+                  width,
+                  height,
+                },
+                normalizedHitSlop,
+                event.changedTouches.at(-1)
+              ) ||
+              isPressedDown.current
+            ) {
+              return;
+            }
+
+            isPressedDown.current = true;
+
+            if (props.unstable_pressDelay) {
+              pressDelayTimeoutRef.current = setTimeout(() => {
+                pressInHandler(event);
+              }, props.unstable_pressDelay);
+            } else {
+              pressInHandler(event);
+            }
+
+            onEndHandlingTouchesDown.current();
+            handlingOnTouchesDown.current = false;
+          });
+        })
+        .onTouchesUp((event) => {
+          if (handlingOnTouchesDown.current) {
+            onEndHandlingTouchesDown.current = () => pressOutHandler(event);
+            return;
+          }
+          pressOutHandler(event);
+        })
+        .onTouchesCancelled((event) => {
+          if (
+            !isPressedDown.current ||
+            event.allTouches.length > event.changedTouches.length
+          ) {
+            return;
+          }
+
+          // original triggers onPressOut on cancel, but not onPress
+          props.onPressOut?.(adaptTouchEvent(event));
+
+          isPressedDown.current = false;
+          setPressedState(false);
+        }),
+    [
+      props.onPress,
+      props.onPressIn,
+      props.onPressOut,
+      setPressedState,
+      isPressedDown,
+      isPressCallbackEnabled,
+      normalizedHitSlop,
+      pressDelayTimeoutRef,
+    ]
+  );
+
+  // rippleGesture lives inside RNButton to enable android's ripple
+  const rippleGesture = useMemo(() => Gesture.Native(), []);
+
+  pressGesture.minDuration(
+    (props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION) +
+      (props.unstable_pressDelay ?? 0)
+  );
+
+  const appliedHitSlop = addInsets(
+    normalizedHitSlop,
+    normalizedPressRetentionOffset
+  );
+
+  const isPressableEnabled = props.disabled !== true;
+
+  touchGesture.enabled(isPressableEnabled);
+  pressGesture.enabled(isPressableEnabled);
+  hoverGesture.enabled(isPressableEnabled);
+  rippleGesture.enabled(isPressableEnabled);
```

Comment:
If it was only enabled it would be fine, but we also have other properties set in that way. Wouldn't it be better to use loop?

```tsx
  const gestures = [touchGesture, pressGesture, hoverGesture, rippleGesture];

  for (const gesture of gestures) {
    gesture.enabled(isPressableEnabled);
    gesture.runOnJS(true);
    gesture.hitSlop(appliedHitSlop);

    if (Platform.OS !== 'web') {
      gesture.shouldCancelWhenOutside(true);
    }
  }

  rippleGesture.hitSlop(normalizedHitSlop);
```

Reply:
Agreed, at first I though wrapping the config into a loop would make it less readable, but now i see this is not the case.